### ISSUE TITLE
Fix duplication of in-page ToC

### ISSUE THREAD

**Issue by atsansone** [flutter/website#9707]
a11y script found duplicate ID for the in-page ToC. The ToC is rendered twice: on [lines 42-45](https://github.com/flutter/website/blob/main/src/_layouts/default.html#L42-L45) for desktop layout and [lines 20-23](https://github.com/flutter/website/blob/main/src/_layouts/default.html#L20-L23) for mobile layout. This raises a [WCAG F77 error](https://squizlabs.github.io/HTML_CodeSniffer/Standards/WCAG2/#sc4_1_1) on pages with in-page ToCs.  This causes issues with accessibility technology. This should be fixed.
